doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,410 @@
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "token": {
+                        "description": "Session token to be used as authentication for further requests.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "token"
+                ],
+                "additionalProperties": false
+            }
+
++ Response 401 (application/json)
+    + Body
+    
+            {
+                "status": 401,
+                "error": "Unauthorized",
+                "message": "You have to supply an username and a password.",
+                "loca": "AUTH_NEED_USERNAME_PASSWORD"
+            }
+            
+    + Schema
+
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "error": {
+                        "description": "Textual description of HTTP status code",
+                        "type": "string"
+                    },
+                    "message": {
+                        "description": "Detailed information about the error.",
+                        "type": "string"
+                    },
+                    "loca": {
+                        "description": "Unique message token that can be used for localization in a frontend application.",
+                        "type": "string"
+                    }
+                },
+                "required": [
+                    "status",
+                    "message",
+                    "loca"
+                ],
+                "additionalProperties": false
+            }
+
+
+## Retrieve current user [GET /user]
+
+Retrieves the currently authenticated users details.
+        
++ Response 200 (application/json)
+    + Body
+    
+            {
+                "status": 200,
+                "username": "the_username",
+                "rank": 1
+            }
+            
+    + Schema
+    
+            {
+                "type": "object",
+                "properties": {
+                    "status": {
+                        "description": "HTTP status code",
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "username": {
+                        "description": "The authenticated users name",
+                        "type": "string"
+                    },
+                    "rank": {
+                        "description": "The authenticated users rank",
```

Comment:
The rank is used for a very basic permission system. It allows to set a rank between 1 and 3, whereas 1 is a user, 2 something like a moderator and 3 an admin. Higher ranks will allow to delete and modify content of other users as well.